fv3core/stencils/dyn_core.py

```diff
@@ -56,12 +71,67 @@ def set_pem(delp: sd, pem: sd, ptop: float):
 
 @gtstencil()
 def heatadjust_temperature_lowlevel(
-    pt: sd, heat_source: sd, delp: sd, pkz: sd, cp_air: float
+    pt: FloatField,
+    heat_source: FloatField,
+    delp: FloatField,
+    pkz: FloatField,
+    cp_air: float,
 ):
     with computation(PARALLEL), interval(...):
         pt[0, 0, 0] = pt + heat_source / (cp_air * delp * pkz)
 
 
+@gtstencil()
+def p_grad_c_stencil(
+    rdxc: FloatField,
+    rdyc: FloatField,
+    uc: FloatField,
+    vc: FloatField,
+    delpc: FloatField,
+    pkc: FloatField,
+    gz: FloatField,
+    dt2: float,
+):
+    """Update C-grid winds from the pressure gradient force
+
+    When this is run the C-grid winds have almost been completely
+    updated by computing the momentum equation terms, but the pressure
+    gradient force term has not yet been applied. This stencil completes
+    the equation and Arakawa C-grid winds have been advected half a timestep
+    upon completing this stencil..
+
+     Args:
+         uc: x-velocity on the C-grid (inout)
+         vc: y-velocity on the C-grid (inout)
+         delpc: vertical delta in pressure (in)
+         pkc:  pressure if non-hydrostatic,
+               (edge pressure)**(moist kappa) if hydrostatic(in)
+         gz:  height of the model grid cells (m)(in)
+         dt2: half a model timestep (for C-grid update) in seconds (in)
+    Grid variable inputs:
+        rdxc, rdyc
+    """
+    from __externals__ import local_ie, local_is, local_je, local_js, namelist
+
+    with computation(PARALLEL), interval(...):
+        if __INLINED(namelist.hydrostatic):
+            wk = pkc[0, 0, 1] - pkc
```

Comment:
Using a [0,0,1] offset inside a full interval here works because the domain is constricted, right? Does this impact the stencil fusion process?

I guess I'm asking if it's preferable to restrict the domain and use `interval(...)` or to use the full domain and have `interval(0,-1)`? The former seems better from a data handling perspective, but the latter seems easier for the backend to fuse. Am I just overthinking this?

Reply:
Good question! I think we don't know yet whether this choice have an impact, though definitely seems worth considering. Our arrays have an extra buffer point in every directions because sometimes they are used, but most of the time 'full_domain' does not include that top buffer point. I think if we have the answer to your question we should do a refactor pass through the dycore as both patterns are abundant.  @eddie-c-davis what do you think?

Reply:
I would say it depends on the surrounding intervals right now. The current fusion heuristic requires the intervals to be the same in order to fuse, and improved fusion probably wins over storage reduction in that case. Would be an interesting experiment though!